crates/services/consensus_module/poa/src/service_test/test_time.rs

```diff
@@ -0,0 +1,81 @@
+use std::time::Duration;
+
+use fuel_core_types::tai64::{
+    Tai64,
+    Tai64N,
+};
+use tokio::{
+    sync::watch,
+    time::Instant,
+};
+
+use crate::ports::GetTime;
+
+pub struct TestTime {
+    time: watch::Sender<Tai64N>,
+    last_observed_tokio_time: Instant,
+}
+
+impl TestTime {
+    pub fn at_unix_epoch() -> Self {
+        Self::new(Tai64N::UNIX_EPOCH)
+    }
+
+    pub fn new(start_time: Tai64N) -> Self {
+        tokio::time::pause();
+        let time = watch::Sender::new(start_time);
+        let last_observed_tokio_time = Instant::now();
```

Comment:
IS there a scenario we'd want this to also be at the `start_time`? Are we okay with this always having a relative dependency on system time `now`?

I guess it's a YAGNI thing. We have what we need for the current tests.

Reply:
The crux here is that there is that `Instant` is only useful for relative time comparisons. Looking at the implementation of `Instant::now()` it boils down to calling the libc `clock_gettime` function with `CLOCK_UPTIME_RAW`, which can't be compared to a timestamp.

I would love to get rid of the system time dependency, but I am not currently aware of a feasible way to do it.